lib/plugins/exchange/ccxt.js

```diff
@@ -8,18 +8,19 @@ const { ORDER_TYPES } = require('./consts')
 const DEFAULT_PRICE_PRECISION = 2
 const DEFAULT_AMOUNT_PRECISION = 8
 
-function trade (side, account, cryptoAtoms, fiatCode, cryptoCode, exchangeName) {
+function trade (side, account, cryptoAtoms, fiatCode, cryptoCode, exchangeName, tradeId) {
   try {
     const exchangeConfig = ALL[exchangeName]
     if (!exchangeConfig) throw Error('Exchange configuration not found')
 
-    const { loadOptions, loadConfig = _.noop, REQUIRED_CONFIG_FIELDS, ORDER_TYPE, AMOUNT_PRECISION } = exchangeConfig
+    const { loadTradeId, loadOptions, loadConfig = _.noop, REQUIRED_CONFIG_FIELDS, ORDER_TYPE, AMOUNT_PRECISION } = exchangeConfig
     if (!isConfigValid(account, REQUIRED_CONFIG_FIELDS)) throw Error('Invalid config')
 
     const symbol = buildMarket(fiatCode, cryptoCode, exchangeName)
     const precision = _.defaultTo(DEFAULT_AMOUNT_PRECISION, AMOUNT_PRECISION)
     const amount = toUnit(cryptoAtoms, cryptoCode).toFixed(precision)
-    const options = _.isFunction(loadOptions) ? loadOptions(account) : {}
+    const accountOptions = _.isFunction(loadOptions) ? loadOptions(account) : {}
+    const options = loadTradeId(accountOptions, tradeId)

```

Comment:
Then you can export CUSTOM_KEY as the id only on the exchanges that support it, so in our case only kraken will export : `CUSTOM_KEY = 'userref'`

```suggestion
    const accountOptions = _.isFunction(loadOptions) ? loadOptions(account) : {}
    
    const withCustomKey = CUSTOM_KEY ? { [CUSTOM_KEY]: tradeId } : {}
    
    const options = _assign(accountOptions, withCustomKey)

```

Reply:
And get rid of loadTradeId?

Reply:
Right, have the exchanges export `CUSTOM_KEY` when needed.